### PR TITLE
Implements TAP TODO directive

### LINKED ISSUE #219: Flaky tests
It would be nice if tape would support the ability to mark a test (or assertion) as flaky. It can be quite a pain when a test that depends on a 3rd party online service blocks the deployment of my own software, because that 3rd party is not cooperating.

### PR DESCRIPTION
**Pull request by ronkorving**
Closes #219 as discussed with @awlayton

### CODE REVIEW

**Comment by ljharb** [comment]
Please revert all the whitespace changes - that's a part of this project's style.

**Comment by ljharb** [comment]
@ronkorving wrt test/comment.js failing - what version of node are you using? It passes for me on my OSX machine.

**Comment by ronkorving** [comment]
I've tried on both Node 4.3.0 and 5.6.0, and comment.js fails. I'll make a separate issue in a bit.

Re. whitespace, it may be useful then to add a http://editorconfig.org file. That would make it much easier to contribute.

**Comment by ronkorving** [comment]
All whitespace changes have been reverted.

**Comment by axit-joost** [comment]
As mentioned in #219 , a very warm and welcome +1 for this.

**Comment by ljharb** [comment]
(fixed the test/comment.js tests in 40be68520fedee580462bfaa7e91651154a65bea btw)

**Comment by ronkorving** [comment]
Appreciated :) (to both of you)

**Comment by awlayton** [comment]
Shouldn't you also add a `test.todo` since there is a `test.skip`?

File: test/todo.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            + '\n'
+            + '1..2\n'
+            + '# tests 2\n'
+            + '# pass  2\n'
+            + '\n'
+            + '# ok\n'
+        )
+    }));
+
+    test("success", function (t) {
```
Comment (ljharb) [inline test/todo.js]: please stick with single-quoted strings, for consistency (same below)

**Comment by ljharb** [comment]
+1 on `test.todo` (and the associated documentation)

File: test/todo.js (hunk 1 above)
Comment (ronkorving) [inline test/todo.js]: Happily :) I think I saw some test names with double quotes elsewhere so kinda went with that. Wasn't really sure what was expected tbh.

**Comment by gabrielcsapo** [comment]
👍

**Comment by ljharb** [comment]
@ronkorving could you check the "allow edits" checkbox on the RHS of the PR?

**Comment by ronkorving** [comment]
@ljharb Done 👍

**Pull request merged**